### PR TITLE
[docs] - Branch Deployments, v2

### PR DESCRIPTION
**Pull request by erinkcochran87**
This PR:

- Updates `cloud_workspace.yaml` > `dagster_cloud.yaml`
- Updates the GitHub Branch Deployments guide:
   - Attempts to clarify when certain steps should occur based on deployment type + when setup occurs (during onboarding or after)
 - Updates the dagster-cloud CLI Branch Deployments guide

### CODE REVIEW

File: docs/content/dagster-cloud/developing-testing/branch-deployments/using-branch-deployments-with-github.mdx [hunk 1]
```diff
@@ -92,7 +143,40 @@ locations:
...
 ```
 
-### Step 4.2: Configure GitHub Action secrets
+### Step 2.3: Configure GitHub Action secrets
+
+In this step, you'll configure the secrets used by the Branch Deployments GitHub Action. The steps and required secrets vary a bit depending on the type of deployment you're using.
+
+Select your deployment type below to view instructions.
+
+<details>
```
Comment (erinkcochran87) [inline docs/content/dagster-cloud/developing-testing/branch-deployments/using-branch-deployments-with-github.mdx:152]: Based on my testing, it seems like this step is required if the user is adding another GH-based code location after they complete onboarding.

I assumed that even if the connected GH repo had a `dagster_cloud.yaml` file, they'd still have to do this part manually, but didn't have time to fully test this. Confirmation needed pls.

File: docs/content/dagster-cloud/developing-testing/branch-deployments/using-branch-deployments-with-github.mdx [hunk 2]
```diff
@@ -92,7 +143,40 @@ locations:
...
+1. In the repository, click the **Settings** tab.
+2. In the **Security** section of the sidebar, click **Secrets > Actions**.
+3. Click **New repository secret**.
+4. In the **Name** field, enter the name of the secret. For example, `DAGSTER_CLOUD_API_TOKEN`
+5. In the **Value** field, paste the value of the secret.
+6. Click **Add secret**.
+
+Repeat steps 3-6 for each of the following:
+
+- `DAGSTER_CLOUD_API_TOKEN` - The Serverless agent token. Refer to the [Managing user and agent tokens guide](/dagster-cloud/account/managing-user-agent-tokens) for more info.
```
Comment (erinkcochran87) [inline docs/content/dagster-cloud/developing-testing/branch-deployments/using-branch-deployments-with-github.mdx:164]: If a Serverless user is doing this post-onboarding (and the GH app setup is no longer available), should they generate a new agent token or use an existing agent token?

**Review by gibsondan — COMMENTED** [review]
oh wow this is still here more than a month later. erin are you still hoping to get this in? @benpankow you're probably the most qualified to confirm the github action parts of this

**Comment by gibsondan** [comment]
I can't figure out how to remove myself as a reviewer and this is very old so I am going to just close it for now to clear my queue, but definitely feel free to re-open if it's still active :)

**Pull request closed without merging**